src/webkitgtk/web_context.rs

```diff
@@ -361,28 +361,33 @@ where
         }
       };
 
-      let request_ = request.clone();
+      let request_ = MainThreadRequest(request.clone());
       let responder: Box<dyn FnOnce(HttpResponse<Cow<'static, [u8]>>)> =
         Box::new(move |http_response| {
-          let buffer = http_response.body();
-          let input = gtk::gio::MemoryInputStream::from_bytes(&gtk::glib::Bytes::from(buffer));
-          let content_type = http_response
-            .headers()
-            .get(CONTENT_TYPE)
-            .and_then(|h| h.to_str().ok());
-
-          let response = URISchemeResponse::new(&input, buffer.len() as i64);
-          response.set_status(http_response.status().as_u16() as u32, None);
-          if let Some(content_type) = content_type {
-            response.set_content_type(content_type);
-          }
+          gtk::glib::idle_add(move || {
```

Comment:
should we use [`MainContext::invoke`](https://docs.rs/glib/0.18.5/glib/struct.MainContext.html#method.invoke) instead? seems like `idle_add` might wait until the loop is idle which may impact performance, also `MainContext::invoke` will call the closure directly if it is already on the main thread.

Reply:
oh i'll try that

Reply:
sorted @amrbashir

Reply:
i knew that function existed - just couldn't find it in tao/wry

Reply:
> sorted @amrbashir

did you forget to push :joy:?

Reply:
🤣🤣🤣🤣 in my defense i was testing this while holding a baby

Reply:
no worries, anything for the baby ❤️

Reply:
finally pushed